core/primitives/src/congestion_info.rs

```diff
@@ -95,6 +93,12 @@ impl CongestionControl {
         }
     }
 
+    pub fn is_fully_congested(congestion_level: f64) -> bool {
```

Comment:
nice

Reply:
but then we'd have to compute the congestion level twice in:
```rust
pub fn outgoing_gas_limit(&self, sender_shard: ShardId) -> Gas {
let congestion = self.congestion_level();

if Self::is_fully_congested(congestion) {
    // Red traffic light: reduce to minimum speed
    if sender_shard == ShardId::from(self.info.allowed_shard()) {
        self.config.allowed_shard_outgoing_gas
    } else {
        0
    }
} else {
    mix(self.config.max_outgoing_gas, self.config.min_outgoing_gas, congestion)
}
}
```